plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -373,17 +374,31 @@ public void writeBlobAtomic(String blobName, InputStream inputStream, long blobS
     }
 
     @Override
-    public DeleteResult delete() {
+    public DeleteResult delete() throws IOException {
         PlainActionFuture<DeleteResult> future = new PlainActionFuture<>();
         deleteAsync(future);
-        return future.actionGet();
+        return getFutureValue(future);
     }
 
     @Override
-    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) {
+    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
         PlainActionFuture<Void> future = new PlainActionFuture<>();
         deleteBlobsAsyncIgnoringIfNotExists(blobNames, future);
-        future.actionGet();
```

Comment:
`PlainActionFuture.actionGet` is already handling `InterruptedException`. The only handling required is of `UncategorizedExecutionException` which is an un-checked exception.